crypto/src/key/hdkd/derivable.rs

```diff
@@ -25,21 +25,44 @@ pub enum DerivationError {
     InvalidChildNumberFormat,
     #[error("Malformed derivation path format")]
     InvalidDerivationPathFormat,
-    #[error("Unsupported derivation type")]
-    UnsupportedDerivationType,
     #[error("Unsupported derivation for key type")]
     UnsupportedKeyType,
     #[error("Key derivation error")]
     KeyDerivationError,
+    #[error("Derivation path too long")]
+    PathTooLong,
+    #[error("Cannot derive path: {0}")]
+    CannotDerivePath(DerivationPath),
+    #[error("Cannot derive hardened key from public key: {0}")]
+    CannotDeriveHardenedKeyFromPublicKey(ChildNumber),
 }
 
 pub trait Derivable: Sized {
-    /// Derive a child private key given a derivation path
+    /// Derive a child private key given a derivation path. The derivation path must include
+    /// the path of this key. For example:
+    /// - If this (self) key has the path m/1/2/3
+    /// - Then the requested path should be m/1/2/3/<rest/of/the/path>
     fn derive_path(self, path: &DerivationPath) -> Result<Self, DerivationError> {
```

Comment:
Given the comment, maybe this could be called `derive_absolute_path`. To disambiguate from a hypothetical function that appends a bunch of components to the existing path.